src/trigger.rs

```diff
@@ -1091,65 +1083,298 @@ pub fn validate_target(token: &Token, data: &Everything, sc: &mut ScopeContext,
+    // change to outscopes
+    let mut outscopes_token = func.clone();
+    outscopes_token.combine(arg, ':');
+    if func.lowercase_is("scope") {
+        sc.replace_named_scope(arg.as_str(), &outscopes_token);
+        if part_flags.contains(PartFlags::Last | PartFlags::Question) {
+            sc.exists_scope(arg.as_str(), outscopes_token);
+        }
+    } else {
+        sc.replace(outscopes, outscopes_token);
+    }
+}
+
+/// Validate that the prefix token does exist and match the `wanted` string,
+/// and that the argument is valid.
+#[cfg(feature = "ck3")]
+pub fn validate_prefix_reference_token(token: &Token, data: &Everything, wanted: &str) {
+    if let Some((prefix, arg)) = token.split_once(':') {
+        let mut sc = ScopeContext::new(Scopes::None, token);
+        if let Some((_, _, trigger)) = scope_prefix(&prefix) {
+            validate_argument_internal(&arg, trigger, data, &mut sc);
+        }
+        if prefix.lowercase_is(wanted) {
+            return;
+        }
+    }
+    let msg = format!("should start with `{wanted}:` here");
+    error(token, ErrorKey::Validation, &msg);
+}
+
+/// Validate that the argument passed through is valid, either being of a complex trigger compare value,
+/// or a scope prefix.
+pub fn validate_argument(
+    part_flags: PartFlags,
+    func: &Token,
+    arg: &Token,
+    data: &Everything,
+    sc: &mut ScopeContext,
+) {
+    let scope_trigger_complex = match Game::game() {
+        #[cfg(feature = "ck3")]
+        Game::Ck3 => crate::ck3::tables::triggers::scope_trigger_complex,
+        #[cfg(feature = "vic3")]
+        Game::Vic3 => crate::vic3::tables::triggers::scope_trigger_complex,
+        #[cfg(feature = "imperator")]
+        // Imperator does not use `()`
+        Game::Imperator => unimplemented!(),
+    };
+
+    let func_lc = func.as_str().to_lowercase();
+    if let Some((inscopes, trigger)) = scope_trigger_complex(&func_lc) {
+        sc.expect(inscopes, &Reason::Token(func.clone()));
+        validate_argument_internal(arg, trigger, data, sc);
+        sc.replace(Scopes::Value, func.clone());
+    } else if let Some(entry) = scope_prefix(func) {
+        validate_argument_scope(part_flags, entry, func, arg, data, sc);
+    } else {
+        let msg = format!("unknown token `{func}`");
+        err(ErrorKey::Validation).msg(msg).loc(func).push();
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::path::PathBuf;
+
+    use crate::{emit_reports, FileKind, Loc};
+
+    use super::*;
+    #[test]
+    fn test_partition() {
+        let path = "./test.txt";
+        let mut loc = Loc::for_file(
+            PathBuf::from_str(path).unwrap(),
+            FileKind::Mod,
+            PathBuf::from_str(path).unwrap(),
+        );
+        loc.line += 1;
+        loc.column += 1;
+        let str = std::fs::read_to_string(path).unwrap();
+        let token = Token::new(&str, loc);
+        let partitioned = partition(&token);
+        dbg!(token);
```

Comment:
Debug output is swallowed by the test framework if the test succeeds. Common practice is to use assert macros to assert that the vector and it's tokens are what's expected.

Reply:
I am more or less using this as a test playground to see if the output (including the error reports) meet my expectation, so kind of difficult to use `assert` here